packages/react/src/components/ExpandCollapsePanel/PanelTrigger.tsx

```diff
@@ -1,55 +1,69 @@
-import React from 'react';
+import React, { forwardRef, Ref } from 'react';
 import PropTypes from 'prop-types';
 import classnames from 'classnames';
 import Icon, { IconType } from '../Icon';
 
 export interface PanelTriggerProps
   extends Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'children'> {
-  children?: ((props: { open: boolean }) => React.ReactNode) | React.ReactNode;
+  children?:
+    | ((props: { open: boolean }) => React.ReactNode)
+    | React.ReactElement;
   open?: boolean;
   fullWidth?: string;
   onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
   iconExpanded?: IconType;
   iconCollapsed?: IconType;
+  hideIcon?: boolean;
+  buttonRef?: Ref<HTMLButtonElement>;
 }
 
-function PanelTrigger({
-  children,
-  className,
-  open,
-  fullWidth,
-  onClick,
-  iconExpanded = 'chevron-down',
-  iconCollapsed = 'chevron-right',
-  ...other
-}: PanelTriggerProps) {
-  return (
-    <button
-      {...other}
-      className={classnames(
-        'ExpandCollapse__trigger',
-        fullWidth ? 'fullWidth' : '',
-        className
-      )}
-      type="button"
-      aria-expanded={open}
-      onClick={onClick}
-    >
-      <div className="ExpandCollapse__trigger-title">
-        {typeof children === 'function' ? children({ open: !!open }) : children}
-      </div>
-      <Icon type={open ? iconExpanded : iconCollapsed} />
-    </button>
-  );
-}
+const PanelTrigger = forwardRef<HTMLButtonElement, PanelTriggerProps>(
+  (
+    {
+      children,
+      className,
+      open,
+      fullWidth,
+      onClick,
+      buttonRef,
+      iconExpanded = 'chevron-down',
+      iconCollapsed = 'chevron-right',
+      hideIcon = false,
+      ...otherProps
+    }: PanelTriggerProps,
+    ref
+  ) => {
+    return (
+      <button
+        className={classnames(
+          className ? className : 'ExpandCollapse__trigger',
+          fullWidth ? 'fullWidth' : '',
+          open ? 'expanded' : ''
```

Comment:
I know some of this was already there, but `classnames` can accept an object with boolean values, example below

```jsx
className={classnames('ExpandCollapse__trigger', className, {
  'fullWidth': fullWidth,
  ...
})
```

Also is the `expanded` class needed? We can use `[aria-expanded="true"]` attribute to achieve the same thing.

Reply:
Oh, cool - I never knew that about `classnames`! Will definitely give that a try and I don't have strong feelings one way or the other, re: the expanded class and you are right that it could use `aria-expanded`.

Reply:
All great points and I agree 100%! Definitely like that approach better and I made that change!